src/tests/execute.spec.ts

```diff
@@ -0,0 +1,19 @@
+import { MiddlewareRegistry } from "../MiddlewareRegistry";
+import { NextApiRequest } from "next";
+
+describe('MiddlewareRegistry.execute', () => {
+    it('should not call a middleware before execute', async () => {
+        const request = { url: '/api/a' } as NextApiRequest;
+        const registry = new MiddlewareRegistry(request);
+        // GIVEN a middleware
+        const middleware = jest.fn();
+        // GIVEN route /api/a
```

Comment:
Another nit, but could you combine these 2 `// GIVEN` comments and move to the top of the test